### PR TITLE
Fix max iteration in Growing Spheres

### PR DESCRIPTION
**Pull request by kmariuszk**

### CODE REVIEW

File: src/generators/non_gradient_based/growing_spheres/growing_spheres.jl [hunk 1]
```diff
@@ -38,7 +38,7 @@ function growing_spheres_generation!(ce::AbstractCounterfactualExplanation)
     factual = ce.x
     counterfactual_data = ce.data
     target = [ce.target]
-    max_iter = 1000
+    max_iter = ce.max_iter
```
Comment (pat-alt) [inline src/generators/non_gradient_based/growing_spheres/growing_spheres.jl:41]: At least with the current struct, we need `ce.convergence[:max_iter]` here (tests currently failing)

**Review by pat-alt — CHANGES_REQUESTED** [review]

File: src/generators/non_gradient_based/growing_spheres/growing_spheres.jl (hunk 1 above)
Comment (kmariuszk) [reply src/generators/non_gradient_based/growing_spheres/growing_spheres.jl]: 👌

**Review by pat-alt — COMMENTED** [review]
Test for benchmarking are failing both for sequential and parallel computation. It's a little hard to tell from the stack trace in CI. Maybe you can try to run this part locally (e.g. using the `docs/benchmarking.qmd` notebook) to figure out what's wrong? Otherwise this looks good

**Comment by kmariuszk** [comment]
> Test for benchmarking are failing both for sequential and parallel computation. It's a little hard to tell from the stack trace in CI. Maybe you can try to run this part locally (e.g. using the `docs/benchmarking.qmd` notebook) to figure out what's wrong? Otherwise this looks good

Either me or @VincentPikand will get back to it once #352 is merged, thanks for the suggestion though!

**Comment by pat-alt** [comment]
> I'm afraid it hasn't been addressed by #352.
> 
> https://github.com/JuliaTrustworthyAI/CounterfactualExplanations.jl/blob/29983f8131d6e161efc938217308e00cf1d41ba7/src/generators/non_gradient_based/growing_spheres/growing_spheres.jl#L41C8-L41C8
> 
> The `max_iter` variable is not properly assigned and thus I'm afraid it's also not used properly.
> 
> We should get back to it once #352 is merged.

Ok, so let's just keep this open for now and as a first step merge changes from main once #355 has been merged?

**Comment by kmariuszk** [comment]
> > I'm afraid it hasn't been addressed by #352.
> > https://github.com/JuliaTrustworthyAI/CounterfactualExplanations.jl/blob/29983f8131d6e161efc938217308e00cf1d41ba7/src/generators/non_gradient_based/growing_spheres/growing_spheres.jl#L41C8-L41C8
> > The `max_iter` variable is not properly assigned and thus I'm afraid it's also not used properly.
> > We should get back to it once #352 is merged.
> 
> Ok, so let's just keep this open for now and as a first step merge changes from main once #355 has been merged?

Yep, I'm on it

**Comment by kmariuszk** [comment]
Opened a new PR: #366

**Pull request closed without merging**